tools/update-ecma-version-tests.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * @fileoverview Generate `<test-name>.result.js` from `<test-name>.src.js` in
+ *               `tests/fixtures/ecma-version/<number>/**` directory.
+ * @author Nicholas C. Zakas
+ * @author Toru Nagashima
+ *
+ * Usage:
+ *      node tools/update-ecma-version-tests.js <number>
+ *
+ */
+
+//------------------------------------------------------------------------------
+// Imports
+//------------------------------------------------------------------------------
+
+import path from "path";
+import { fileURLToPath } from "url";
+import util from "util";
+import shelljs from "shelljs";
+import * as espree from "../espree.js";
+import tester from "../tests/lib/tester.js";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get the parse result.
+ * @param {string} code The source code to get.
+ * @param {Object} config The espree options.
+ * @returns {string} The parse result.
+ */
+function getExpectedResult(code, config) {
+    try {
+        return tester.getRaw(espree.parse(code, config));
+    } catch (ex) {
+        const raw = tester.getRaw(ex);
+
+        raw.message = ex.message;
+        return raw;
+    }
+}
+
+/**
+ * Find the test cases from a given directory.
+ * Tests are the name of files that end with `.src.js`.
+ * @param {string} directory The root directory to find.
+ * @returns {string[]} The found test names.
+ */
+function findTests(directory) {
+    return shelljs.find(directory)
+        .filter(filename => filename.endsWith(".src.js"))
+        .map(filename => filename.slice(0, -".src.js".length)); // strip off ".src.js"
+}
+
+/**
+ * Write the parse result to the result file (`<test-name>.result.js`).
+ * @param {Object} result The parse result to write.
+ * @param {string} testResultFilename The path to the result file.
+ * @returns {void}
+ */
+function outputResult(result, testResultFilename) {
+    let code = `export default ${JSON.stringify(result, (key, value) => ((typeof value === "bigint") ? `bigint<${value}n>` : value), 4)};`;
+
```

Comment:
`undefined`, `NaN` (and bigint) are not supported in JSON.

Reply:
I was wrong. `Infinity` can appear in AST such as `1e123345`